cluster-operator/src/main/java/io/strimzi/operator/cluster/model/JmxTrans.java

```diff
@@ -0,0 +1,391 @@
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+
+/*
+ * Class for handling JmxTrans configuration passed by the user. Used to get the resources needed to create the
+ * JmxTrans deployment including: config map, deployment, and service accounts.
+ */
+public class JmxTrans extends AbstractModel {
+
+    // Configuration defaults
+    private static final String STRIMZI_DEFAULT_JMXTRANS_IMAGE = "STRIMZI_DEFAULT_JMXTRANS_IMAGE";
+    public static final Probe READINESS_PROBE_OPTIONS = new ProbeBuilder().withTimeoutSeconds(5).withInitialDelaySeconds(15).build();
+    private static final io.strimzi.api.kafka.model.Probe DEFAULT_JMX_TRANS_PROBE = new io.strimzi.api.kafka.model.ProbeBuilder()
+            .withInitialDelaySeconds(JmxTransSpec.DEFAULT_HEALTHCHECK_DELAY)
+            .withTimeoutSeconds(JmxTransSpec.DEFAULT_HEALTHCHECK_TIMEOUT)
+            .build();
+
+    // Configuration for mounting `config.json` to be used as Config during run time of the JmxTrans
+    public static final String JMXTRANS_CONFIGMAP_KEY = "config.json";
+    public static final String JMXTRANS_VOLUME_NAME = "jmx-config";
+    public static final String CONFIG_MAP_ANNOTATION_KEY = "config-map-revision";
+    protected static final String JMX_METRICS_CONFIG_SUFFIX = "-jmxtrans-config";
+    public static final String JMX_FILE_PATH = "/var/lib/jmxtrans";
+
+    private boolean isDeployed;
+    private boolean isJmxAuthenticated;
+    private String configMapName;
+    private String clusterName;
+
+    /**
+     * Constructor
+     *
+     * @param namespace Kubernetes/OpenShift namespace where JmxTrans resources are going to be created
+     * @param kafkaCluster kafkaCluster name
+     * @param labels    labels to add to the kafkaCluster
+     */
+    protected JmxTrans(String namespace, String kafkaCluster, Labels labels) {
+        super(namespace, kafkaCluster, labels);
+        this.name = JmxTransResources.deploymentName(kafkaCluster);
+        this.clusterName = kafkaCluster;
+        this.replicas = 1;
+        this.readinessPath = "/metrics";
+        this.livenessPath = "/metrics";
+        this.readinessProbeOptions = READINESS_PROBE_OPTIONS;
+
+        this.mountPath = "/var/lib/kafka";
+
+        // Metrics must be enabled as JmxTrans is all about gathering JMX metrics from the Kafka brokers and pushing it to remote sources.
+        this.isMetricsEnabled = true;
+    }
+
+    public static JmxTrans fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup versions) {
+        JmxTrans result = null;
+        JmxTransSpec spec = kafkaAssembly.getSpec().getJmxTrans();
+        if (spec != null) {
+            if (kafkaAssembly.getSpec().getKafka().getJmxOptions() == null) {
+                log.info("Can't start up JmxTrans as Kafka JmxOptions is disabled");
```

Comment:
The reader of the log (i.e. the user) won't know about the Java class `JmxOptions`. It's better to use the Json path `spec.kafka.jmxOptions`, which is how we refer to parts on the CR within the user documentation.

So something like:

               log.info("Can't start up JmxTrans as Kafka spec.kafka.jmxOptions is not specified");

But you should also include the name and namespace of the CR, since the operator might be operating on multiple namespaces.